core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -4315,6 +4315,192 @@ class KafkaApisTest extends Logging {
+    when(sharePartitionManager.newContext(any(), any(), any(), any())).thenReturn(
+      new SharePartitionManager.SessionlessShareFetchContext(Map(topicIdPartition -> new ShareFetchRequest.SharePartitionData(topicId, 1000, Optional.empty())).asJava)
+    )
+
+    when(clientQuotaManager.maybeRecordAndGetThrottleTimeMs(
+      any[RequestChannel.Request](), anyDouble, anyLong)).thenReturn(0)
+
+    val shareFetchRequestData = new ShareFetchRequestData().
+      setGroupId("group").
+      setMemberId(memberId.toString).
+      setTopics(List(new ShareFetchRequestData.FetchTopic().
+        setTopicId(topicId).
+        setPartitions(List(
+          new ShareFetchRequestData.FetchPartition().
+            setPartitionIndex(0).
+            setCurrentLeaderEpoch(1)).asJava)).asJava)
+
+    val shareFetchRequest = new ShareFetchRequest.Builder(shareFetchRequestData).build(ApiKeys.SHARE_FETCH.latestVersion)
+    val request = buildRequest(shareFetchRequest)
+    kafkaApis = createKafkaApis()
+    kafkaApis.handleShareFetchRequest(request)
+    val response = verifyNoThrottling[ShareFetchResponse](request)
+    val responseData = response.data()
+
+    assertEquals(0, responseData.errorCode())
+    assertEquals(1, responseData.responses().size())
+    assertEquals(topicId, responseData.responses().get(0).topicId())
+    assertEquals(1, responseData.responses().get(0).partitions().size())
+    assertEquals(partitionIndex, responseData.responses().get(0).partitions().get(0).partitionIndex())
+    assertEquals(100, responseData.responses().get(0).partitions().get(0).errorCode())
+  }
+
+  @Test
+  def testHandleShareFetchRequestErrorInReadingPartition(): Unit = {
+    val topicId = Uuid.randomUuid()
+    val partitionIndex = 0
+    val topicName = "foo"
+    addTopicToMetadataCache(topicName, 1, topicId = topicId)
+    val memberId : Uuid = Uuid.ZERO_UUID
+
+    val topicIdPartition : TopicIdPartition = new TopicIdPartition(topicId, new TopicPartition(topicName, partitionIndex))
+    when(sharePartitionManager.newContext(any(), any(), any(), any())).thenReturn(
+      new SharePartitionManager.SessionlessShareFetchContext(Map(topicIdPartition -> new ShareFetchRequest.SharePartitionData(topicId, 1000, Optional.empty())).asJava)
+    )
+
+    when(clientQuotaManager.maybeRecordAndGetThrottleTimeMs(
+      any[RequestChannel.Request](), anyDouble, anyLong)).thenReturn(0)
+
+    when(sharePartitionManager.fetchMessages(
+      any[FetchParams],
+      any[util.List[TopicIdPartition]],
+      anyString()
+    )).thenReturn(CompletableFuture.completedFuture(Map[TopicIdPartition, ShareFetchResponseData.PartitionData](
+      new TopicIdPartition(topicId, new TopicPartition(topicName, partitionIndex)) ->
+        new ShareFetchResponseData.PartitionData().setPartitionIndex(partitionIndex).setErrorCode(2).setAcknowledgeErrorCode(0).setRecords(null)
```

Comment:
I'd prefer to see proper error codes used here, such as `Errors.NONE` or `Errors.REPLICA_NOT_AVAILABLE`.

Reply:
That makes a lot more sense. Thanks for pointing it out ! I made the changes in the most recent commit